### ISSUE TITLE
[kube-state-metrics] Scraping target broken after update to v4.30

### ISSUE THREAD

**Issue by ldunkum** [prometheus-community/helm-charts#3101]
### Describe the bug a clear and concise description of what the bug is.

We use the kube-prometheus-stack chart and in v45.2.0, the kube-state-metrics was updated to v4.30.*.  
Since the update, the kube-state-metrics Endpoint is in state down with the following error `proto: wrong wireType = 0 for field Metric`.  
As v4.30 introduces kube-state-metrics v2.8.0 (app version), on the first glance it seems like the bug might be related to the feature introduced in https://github.com/kubernetes/kube-state-metrics/pull/1974. Unfortunately, I'm not quite sure, which exact chart version this is happening in, but I've tried it two weeks ago and today, and get the same error, so it's probably happening in multiple chart versions.

I also opened an issue for the kube-prometheus-stack (https://github.com/prometheus-community/helm-charts/issues/3050) but haven't gotten an answer so far. If it is more fitting there, this issue can be closed, but it has been open without any response for about 2 weeks.

### What's your helm version?

v3.11.0 (fluxcd helm-controller)

### What's your kubectl version?

-

### Which chart?

kube-state-metrics

### What's the chart version?

v4.30.*

### What happened?

We updated the kube-prometheus-stack chart version from `v45.1.1` to `v45.2.0`, which resulted in a kube-state-metrics chart update from `v4.24.*` to `v4.30.*`. 

### How to reproduce it?

Update kube-prometheus-stack from v45.1.1 to >= v45.2.0.

### Enter the changed values of values.yaml?

```yaml
resources:
  requests:
    cpu: 10m
    memory: 32Mi
  limits:
    cpu: null
    memory: 128Mi
```

### Enter the command that you execute and failing/misfunctioning.

We use Flux CD, I think it probably runs `helm upgrade` after updating the CRDs, but I'm not exactly sure.

**Comment by CatherineF-dev** [comment]
FYI: it's fixed in v2.8.2

**Comment by CatherineF-dev** [comment]
Please try https://github.com/prometheus-community/helm-charts/pull/3137
